pkg/acme/accounts/registry.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2020 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package accounts
+
+import (
+	"crypto/rsa"
+	"crypto/tls"
+	"errors"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	acmeapi "golang.org/x/crypto/acme"
+
+	acmecl "github.com/jetstack/cert-manager/pkg/acme/client"
+	cmacme "github.com/jetstack/cert-manager/pkg/apis/acme/v1alpha2"
+	"github.com/jetstack/cert-manager/pkg/util"
+)
+
+// ErrNotFound is returned by GetClient if there is no ACME client registered.
+var ErrNotFound = errors.New("ACME client for issuer not initialised/available")
+
+// A registry provides a means to store and access ACME clients using an issuer
+// objects UID.
+// This is used as a shared cache of ACME clients across various controllers.
+type Registry interface {
+	// AddClient will ensure the registry has a stored ACME client for the Issuer
+	// object with the given UID, configuration and private key.
+	AddClient(uid string, config cmacme.ACMEIssuer, privateKey *rsa.PrivateKey)
```

Comment:
Do we support ECDSA key accounts?

Reply:
no, although we could in future, I opted to not add it in for now as it will require API changes

Reply:
Sgtm